src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -398,6 +406,95 @@ def destroy(self, request, *args, **kwargs):
                 status=HTTPStatus.FORBIDDEN,
             )
 
+    @extend_schema(
+        request=None,
+        responses={
+            (
+                201,
+                "application/json",
+            ): serializers.AdminCacheGenerateCertificateSerializer,
+            (
+                202,
+                "application/json",
+            ): serializers.AdminCacheGenerateCertificateSerializer,
+            400: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(
+        methods=["POST"],
+        detail=True,
+    )
+    def generate_certificates(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Generate the certificates for a course product relation when it is eligible.
+        """
+        course_product_relation = self.get_object()
+        cache_key = f"celery_certificate_generation_{course_product_relation.id}"
+
+        if cache_data := cache.get(cache_key):
+            return JsonResponse(
+                serializers.AdminCacheGenerateCertificateSerializer(cache_data).data,
```

Comment:
I don't see the point to use a serializer here ! Your data is from the beginning a valid dict serializable in json when returning a `JsonResponse`.